code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/MessagingFullscreenEventListener.java

```diff
@@ -83,9 +86,41 @@ public void onDismiss(@NonNull final Presentable<InAppMessage> fullscreenMessage
     public void onError(
             @NonNull final Presentable<InAppMessage> presentable,
             @NonNull final PresentationError presentationError) {
+        PresentableMessageMapper.InternalMessage message =
+                (PresentableMessageMapper.InternalMessage)
+                        PresentableMessageMapper.getInstance()
+                                .getMessageFromPresentableId(presentable.getPresentation().getId());
+        if (message != null) {
+            final String errorReason = getErrorReason(presentationError);
+            if (!StringUtils.isNullOrEmpty(errorReason) && message.getAutoTrack()) {
+                message.track(errorReason, MessagingEdgeEventType.SUPPRESSED_DISPLAY);
+            }
+            message.recordEventHistory(errorReason, MessagingEdgeEventType.SUPPRESSED_DISPLAY);
+        }
+
         Log.debug(MessagingConstants.LOG_TAG, SELF_TAG, "Fullscreen message failed to show.");
     }
 
+    /**
+     * Returns the error reason from the {@code PresentationError} instance. For now we are only
+     * interested in {@link SuppressedByAppDeveloper} and {@link ConflictingPresentation}. An empty
+     * string will be returned for other {@code PresentationError} types.
+     *
+     * @param presentationError the {@link PresentationError} which occurred
+     * @return the error reason {@code String} if available, otherwise null
+     */
+    private static @Nullable String getErrorReason(
+            @NonNull final PresentationError presentationError) {
+        final Class presentationErrorClass = presentationError.getClass();
+        String errorReason = null;
+        if (presentationErrorClass.equals(SuppressedByAppDeveloper.class)) {
```

Comment:
can we do `if(presentationError instanceOf SuppressedByAppDeveloper)` here?

Reply:
good catch, updated this